irods_specific_implementation/irods_rest_logical_path_api_implementation.h

```diff
@@ -115,10 +117,429 @@ namespace irods::rest
                 error("Caught exception - {}", e.what());
                 return make_error_response(SYS_INTERNAL_ERR, e.what());
             }
-            catch (...) {
-                error("Caught unknown exception");
-                return make_error_response(SYS_UNKNOWN_ERROR, "Unknown error occurred");
+	    catch (...){
+		error("Caught unknown exception.");
+		return make_error_response(SYS_UNKNOWN_ERROR, "Caught unknown exception");
+	    }
+	}
+
+        std::tuple<Pistache::Http::Code, std::string>
+        trim_dispatcher(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+
+            try {
+              auto conn = get_connection(_request.headers().getRaw("authorization").value());
+              dataObjInp_t trim_input{};
+
+              bool is_collection = false;
+
+              setup_input_for_trim(
+                _request,
+                &trim_input,
+                *conn(),
+                is_collection
+              );
+
+              if ( getValByKey(&trim_input.condInput, REPL_NUM_KW) != nullptr &&
+                   getValByKey(&trim_input.condInput, RESC_NAME_KW) != nullptr
+                 )
+              {
+                return std::make_tuple(
+                  Pistache::Http::Code::Bad_Request,
+                  "Both src-resc and repl-num are specified. These are incompatible. See https://docs.irods.org/4.3.0/icommands/user/#itrim"
+                );
+              }
```

Comment:
This endpoint doesn't need to check for incompatible options. We can rely on the server to do this for us.

The REST API is primarily just a shim/pass-through and so it doesn't need to interpret anything. We can leave interpretation of error info to the client interacting with the REST API.

These statements apply to the rest of this file as well.

Reply:
Even if the server will catch bad input for us, do we not want to catch anything that can be caught "statically" before it sends off an API request to the server? My thinking was similar to the situation with checking whether `recursive=0` while targeting a collection: the cost to check would be offset by the reduction in traffic to the server.

Reply:
Doing the same checks as the server means they can become out-of-sync.

You are correct that this can lead to more network traffic, but that will only happen when bad input is provided. This application is a shim around the iRODS API. Consumers of it will tweak things until they know the invocations are correct. Once things work as expected, having these checks only slow down the operations because we're doing them twice.

Keep in mind that this is a programming interface meant for developers.

Reply:
Thinking about it in terms of the use-case does make that clear. Should I remove `recursive=0` check in the delete endpoint as well?

Reply:
Yes. Make your job easy and let the server take care of the heavy lifting.

Reply:
Actually, the implementation looks like it will not get caught if we try to target a collection without recursive, since the only actually communication with the server is rsDataObjRepl, on the individual data objects...

Reply:
Looks like this has been addressed, so I am going to resolve this.

Reply:
It looks like this has been addressed, so I am going to resolve this.